bm_soft_data_msg.cpp

```diff
@@ -0,0 +1,141 @@
+#include "bm_soft_data_msg.h"
+
+CborError BmSoftDataMsg::encode(Data &d, uint8_t *cbor_buffer, size_t size,
+                                size_t *encoded_len) {
+  CborError err;
+  CborEncoder encoder, map_encoder;
+  cbor_encoder_init(&encoder, cbor_buffer, size, 0);
+
+  do {
+    err = cbor_encoder_create_map(&encoder, &map_encoder, NUM_FIELDS);
+    if (err != CborNoError) {
+      printf("cbor_encoder_create_map failed: %d\n", err);
+      if (err != CborErrorOutOfMemory) {
+        break;
+      }
+    }
+
+    // sensor_header_msg
+    err = SensorHeaderMsg::encode(map_encoder, d.header);
+    if (err != CborNoError) {
+      printf("SensorHeaderMsg::encode failed: %d\n", err);
+      if (err != CborErrorOutOfMemory) {
+        break;
+      }
+    }
+
+    // temperature_deg_c
+    err = cbor_encode_text_stringz(&map_encoder, "temperature_deg_c");
+    if (err != CborNoError) {
+      printf("cbor_encode_text_stringz failed for temperature_deg_c key: %d\n",
+             err);
+      if (err != CborErrorOutOfMemory) {
+        break;
+      }
+    }
+    err = cbor_encode_double(&map_encoder, d.temperature_deg_c);
+    if (err != CborNoError) {
+      printf("cbor_encode_double failed for temperature_deg_c value: %d\n",
+             err);
+      if (err != CborErrorOutOfMemory) {
+        break;
+      }
+    }
+
+    if (err != CborNoError && err != CborErrorOutOfMemory) {
+      break;
+    }
```

Comment:
I think this can be removed as redundant with the break above. Let me know if I'm missing something.